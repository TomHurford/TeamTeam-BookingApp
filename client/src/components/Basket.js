import React, {Component} from 'react';
import '../styles/TitleOfPage.css';
import PropTypes from 'prop-types';

class Basket extends Component {
  render() {
    return (
    <div>  
      <h1 className='title'> Basket </h1>
    <ol className="basket">
      {this.props.tickets && this.props.tickets.map((ticket) => (
        <li key={ticket.id}>
          <div className="basket-item">
            <p  className="basket-item-event-name">{ticket.ticket.name}</p>
            <p  className="basket-item-type">{ticket.ticket.type}</p>
            <p  className="basket-item-price">$ {ticket.ticket.price}</p>
            <button className="basket-item-remove"  data-testid="remove-button" onClick={() => this.props.removeTicket(ticket)}>Remove</button>
          </div>
        </li>
      ))}
    </ol>
    <h1>Total: ${this.props.totalPrice}</h1>
    </div>
    );
  }
}

Basket.propTypes = {
  tickets: PropTypes.array,
  totalPrice: PropTypes.number,
  removeTicket: PropTypes.func
};
export default Basket;