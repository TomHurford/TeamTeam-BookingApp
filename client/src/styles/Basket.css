.basket-container {
    padding-top: 20px;
    width: 100%;
    display: none;
}
.basket-container > .header {
    width: 100%;
}
.basket-container > .header > h1.title {
    font-size: 2em;
    color: black;
    text-align: center;
}
.basket-container > .body {
    
}
.basket-container > .footer {
 
}

#active {
    display: block;
}
.empty-basket > .body {
    text-align: center;
}