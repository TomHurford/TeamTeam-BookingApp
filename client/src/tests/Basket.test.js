import React from 'react';
import renderer from 'react-test-renderer';
import Basket from '../components/Basket';

test('Basket snapshot test', () => {
  const component = renderer.create(<Basket />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});