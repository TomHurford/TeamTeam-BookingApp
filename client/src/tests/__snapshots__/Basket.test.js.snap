// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basket snapshot test 1`] = `
<div>
  <h1
    className="title"
  >
     Basket 
  </h1>
  <ol
    className="basket"
  />
  <h1>
    Total: $
  </h1>
</div>
`;

exports[`Basket with tickets snapshot test 1`] = `
<div>
  <h1
    className="title"
  >
     Basket 
  </h1>
  <ol
    className="basket"
  >
    <li>
      <div
        className="basket-item"
      >
        <p
          className="basket-item-event-name"
        >
          porro dolores magni
        </p>
        <p
          className="basket-item-type"
        >
          PAID
        </p>
        <p
          className="basket-item-price"
        >
          $ 
          10
        </p>
        <button
          className="basket-item-remove"
          data-testid="remove-button"
          onClick={[Function]}
        >
          Remove
        </button>
      </div>
    </li>
  </ol>
  <h1>
    Total: $
    10
  </h1>
</div>
`;
